mobile/src/main/java/org/openhab/habdroid/model/NfcTag.java

```diff
@@ -0,0 +1,76 @@
+package org.openhab.habdroid.model;
+
+import android.net.Uri;
+import android.text.TextUtils;
+import androidx.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import org.openhab.habdroid.background.BackgroundTasksManager;
+import org.openhab.habdroid.ui.WriteTagActivity;
+
+import static org.openhab.habdroid.background.BackgroundTasksManager.WORKER_TAG_PREFIX_NFC;
+
+@AutoValue
+public abstract class NfcTag {
+    @Nullable
+    public abstract String sitemap();
+    @Nullable
+    public abstract String item();
+    @Nullable
+    public abstract String label();
+    @Nullable
+    public abstract String state();
+    @Nullable
+    public abstract String mappedState();
+    public abstract boolean isOpenHabTag();
+    public abstract boolean isSitemapTag();
+    public abstract boolean hasLabelAndMappedState();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+        abstract Builder sitemap(String sitemap);
+        abstract Builder item(String item);
+        abstract Builder label(String label);
+        abstract Builder state(String state);
+        abstract Builder mappedState(String mappedState);
+        abstract Builder isOpenHabTag(boolean isOpenHabTag);
+        abstract Builder isSitemapTag(boolean isSitemapTag);
+        abstract Builder hasLabelAndMappedState(boolean hasLabelAndMappedState);
+
+        abstract NfcTag build();
+    }
+
+    public static NfcTag fromTagData(Uri uri) {
```

Comment:
Could just return `null` here if scheme isn't 'openhab'. That way the `isOpenhabTag` method would become superfluous.